### ISSUE TITLE
Add driver toggleable slow-mode

### ISSUE THREAD

**Issue by will-hou** [SouthEugeneRoboticsTeam/InfiniteRecharge-2020#26]
Two parts:

- It would be good to have a customizable `driveSpeedScalar` to modify the speed of the drivetrain with on Shuffleboard. Currently it's just a boolean that halves the drivespeed when true. This can just be done by modifying (and renaming) the existing `TableEntry` [here ](https://github.com/SouthEugeneRoboticsTeam/InfiniteRecharge-2020/blob/d6bb358baef8ff26f24ee2491a7cbf1954c4d7ed/src/main/kotlin/org/sert2521/infiniterecharge2020/drivetrain/DrivetrainPreferences.kt#L16) and setting the `initialValue` to a Double instead of a Boolean. You'd also have to multiply the driver-inputted throttle value by it [here](https://github.com/SouthEugeneRoboticsTeam/InfiniteRecharge-2020/blob/d6bb358baef8ff26f24ee2491a7cbf1954c4d7ed/src/main/kotlin/org/sert2521/infiniterecharge2020/drivetrain/DrivetrainCommands.kt#L45)
- During a match, the driver would not have time to go into shuffleboard and type in a drivespeed scalar to enter slowmode. There should be a button that sets the drivespeed scalar to a predefined value in code (0.5 or something should be good). A function in the `OI.kt` file should be created that sets the value of `driveSpeedScalar` to a predefined value in `drivetrainPreferences` that's triggered by the pressing of the X button on the primary controller. The function should also set the speed scalar back to 1.0 when the button is pressed again. See [here ](https://github.com/SouthEugeneRoboticsTeam/InfiniteRecharge-2020/blob/d6bb358baef8ff26f24ee2491a7cbf1954c4d7ed/src/main/kotlin/org/sert2521/infiniterecharge2020/drivetrain/Drivetrain.kt#L135)for an example of how to set the value of a `TableEntry`

**Comment by andrewda** [comment]
For reference, here's how we handled that last year:

https://github.com/SouthEugeneRoboticsTeam/DeepSpace-2019/blob/development/src/main/kotlin/org/sert2521/deepspace/drivetrain/DrivetrainCommands.kt#L42-L62 (the trigger on the XBox controller allowed variable downscaling while held in).

**Comment by will-hou** [comment]
Oh hm I didn't realize that we already had a `driveSpeedScalar` in the code. Would it be better to modify that or a `TableEntry?` Values in the `Preferences` table persist after rebooting right? I'm worried that we'll accidentally forget to change the value back to 1.0 before the match and we go 🐢🐢. Have we run into that before or is that not really a problem?

**Comment by andrewda** [comment]
@will-hou yeah we ran into that once or twice last year. I think eventually we just set the scalar to 1 on boot.
